test/e2e/page-objects/pages/home/asset-list.ts

```diff
@@ -103,6 +137,42 @@ class AssetListPage {
     return assets.length;
   }
 
+  async getTokenListNames(): Promise<string[]> {
+    console.log(`Retrieving the list of token names`);
+    const tokenElements = await this.driver.findElements(this.tokenListItem);
+    const tokenNames = await Promise.all(
+      tokenElements.map(async (element) => {
+        return await element.getText();
+      }),
+    );
+    return tokenNames;
+  }
+
+  async getAssetPercentageIncreaseDecrease(
+    assetAddress: string,
+  ): Promise<string> {
+    console.log(
+      `Retrieving the percentage increase or decrease for ${assetAddress}`,
+    );
+    const percentageElement = await this.driver.findElement(
+      this.tokenPercentage(assetAddress),
+    );
+    const percentage = await percentageElement.getText();
```

Comment:
mmm this is an anti-pattern that opens the door to race conditions, where the element is rendered but it does not have yet the value we expect.
We should try to avoid finding an element and then getting its text where possible. I think in this case it's possible to avoid this. In the specs below, we should directly try to find the element by its expected inner value instead of this assert

```
  const percentage =
          await assetListPage.getAssetPercentageIncreaseDecrease(tokenAddress);
        assert.equal(percentage, '');
 ```

What do you think?

Reply:
Good point! I agree with you here. I have used `isElementPresent` to check for the existence of the element for a given token address